src/tools/run-make-support/src/rustdoc.rs

```diff
@@ -30,12 +32,14 @@ fn setup_common() -> Command {
 
 impl Rustdoc {
     /// Construct a bare `rustdoc` invocation.
+    #[track_caller]
```

Comment:
Does this `#[track_caller]` really do anything? Is it required if you called e.g. `bare()` and then immediately dropped the result, to get a better error from the drop bomb?

Reply:
Oh this is because I forgot a `#[track_caller]` on `bare_rustdoc()`. Oops, good catch.

Reply:
To be honest, I meant it in a completely different way :laughing: It seemed to me that most of the added `#[track_caller]`s in this PR don't do anything, I just picked this up at random when I sent the comment. That's what I was asking about.

Reply:
Let me go double check if `#[track_caller]` works how I think it works

Reply:
E.g. [here](https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=9b6de795be0dcb27e7a6ab8c787c47c4), commenting out the track caller doesn't seem to change anything.

Reply:
Yes: the answer is that this `#[track_caller]` is actually load-bearing for the panic message's line number:

Test rmake.rs:

```
use run_make_support::bare_rustdoc;

fn main() {
    bare_rustdoc();
}
```

produces

```
thread 'main' panicked at src/tools/run-make-support/src/drop_bomb/mod.rs:43:13:
command constructed but not executed at line 4: `/home/gh-jieyouxu/rust/build/aarch64-unknown-linux-gnu/stage1/bin/rustdoc`
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

without the `#[track_caller]` on the `bare_rustdoc` and transitively `Rustdoc::bare`, the line number is for rustdoc.rs inside the support library

```
thread 'main' panicked at src/tools/run-make-support/src/drop_bomb/mod.rs:43:13:
command constructed but not executed at line 10: `/home/gh-jieyouxu/rust/build/aarch64-unknown-linux-gnu/stage1/bin/rustdoc`
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

Reply:
My understanding is that `#[track_caller]` is a *hint* for when `Location::caller()` is called to traverse up the callstack until an item body (or whatever the correct term is) who is not annotated with `#[track_caller]` is found.

Reply:
Ok, let's keep it then, thanks :)